---
title: "Genome-wide identification and classification of transcription factors in plant genomes"
author: 
  - name: Fabricio Almeida-Silva
    affiliation: |
      VIB-UGent Center for Plant Systems Biology, Ghent University, 
      Ghent, Belgium
  - name: Yves Van de Peer
    affiliation: |
      VIB-UGent Center for Plant Systems Biology, Ghent University, 
      Ghent, Belgium
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_depth: 2
    number_sections: yes
date: "`r Sys.Date()`"
bibliography: vignette_bibliography.bib
vignette: >
  %\VignetteIndexEntry{Genome-wide identification and classification of transcription factors in plant genomes}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL ## Related to https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016656.html
)
```


# Introduction

Transcription factors (TFs) are proteins that bind to cis-regulatory elements 
in promoter regions of genes and regulate their expression. 
Identifying them in a genome is useful for a variety of reasons, such as 
exploring their evolutionary history across clades and inferring 
gene regulatory networks. `r BiocStyle::Githubpkg("almeidasilvaf/tfhunter")` 
allows users to identify plant TFs from whole-genome protein sequences and 
classify them into families and subfamilies (when applicable) using the 
classification scheme implemented in [PlantTFDB](http://planttfdb.gao-lab.org/).

# Installation

You can install `r BiocStyle::Githubpkg("almeidasilvaf/tfhunter")` 
with the following code:

```{r "install", eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
      install.packages("BiocManager")
  }

BiocManager::install("tfhunter")
```

Loading package after installation:

```{r load_package}
library(tfhunter)
```

# Data description

In this vignette, we will use protein sequences of TFs from the algae 
species *Galdieria sulphuraria* as an example. The proteome file was 
downloaded from the PLAZA Diatoms database [@osuna2020seminavis], and 
it was filtered to keep only TFs for demonstration purposes. 
The object `gsu` stores the protein sequences in an `AAStringSet` object.

```{r data}
data(gsu)
gsu
```


# Algorithm description

TF identification and classification is based on the presence of signature 
protein domains, which are identified using profile hidden Markov models (HMMs). 
The family classification scheme is the same as the one used by 
PlantTFDB [@jin2016planttfdb], and it is summarized below: [^1]

```{r scheme, echo = FALSE}
data(classification_scheme)
knitr::kable(classification_scheme)
```


[^1]: **Tip:** You can access this classification scheme in your R session by 
loading the data frame `data(classification_scheme)`.

# Identifying and classifying TFs

To identify TFs from protein sequence data, you will use the 
function `annotate_pfam()`. This function takes as input an `AAStringSet` 
object [^2] and returns a data frame of protein domains associated with 
each sequence. The HMMER program [@finn2011hmmer] is used to scan protein 
sequences for the presence of DNA-binding protein domains, as well as
auxiliary and forbidden domains. Pre-built HMM profiles can be found in the
*extdata/* directory of this package.


[^2]: **Tip:** If you have protein sequences in a FASTA file, you can read 
them into an `AAStringSet` object with the function `readAAStringSet()` from 
the `r BiocStyle::Biocpkg("Biostrings")` package.

```{r identifying_tfs}
data(gsu_annotation)

# Annotate TF-related domains using a local installation of HMMER
if(hmmer_is_installed()) {
  gsu_annotation <- annotate_pfam(gsu)
} 

# Take a look at the first few lines of the output
head(gsu_annotation)
```

Now that we have our TF-related domains, we can classify TFs in families 
with the function `classify_tfs()`.

```{r classifying_tfs}
data(gsu_families)

# Classify TFs into families
if(hmmer_is_installed()) {
  gsu_families <- classify_tfs(gsu_annotation)
} 

# Take a look at the output
head(gsu_families)

# Count number of TFs per family
table(gsu_families$Family)
```

# Session information {.unnumbered}

This document was created under the following conditions:

```{r session_info}
sessioninfo::session_info()
```

# References {.unnumbered}

