<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tfhunter">
<title>Genome-wide identification and classification of transcription factors in plant genomes • tfhunter</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.2/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genome-wide identification and classification of transcription factors in plant genomes">
<meta property="og:description" content="tfhunter">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tfhunter</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_tfhunter.html">Genome-wide identification and classification of transcription factors in plant genomes</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/tfhunter/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genome-wide identification and classification of transcription factors in plant genomes</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      <p>VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent,
Belgium</p>
<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      <p>VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent, Belgium</p>
<br><h4 data-toc-skip class="date">2022-10-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/tfhunter/blob/HEAD/vignettes/vignette_tfhunter.Rmd" class="external-link"><code>vignettes/vignette_tfhunter.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_tfhunter.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Transcription factors (TFs) are proteins that bind to cis-regulatory
elements in promoter regions of genes and regulate their expression.
Identifying them in a genome is useful for a variety of reasons, such as
exploring their evolutionary history across clades and inferring gene
regulatory networks. <em><a href="https://github.com/almeidasilvaf/tfhunter" class="external-link">tfhunter</a></em>
allows users to identify plant TFs from whole-genome protein sequences
and classify them into families and subfamilies (when applicable) using
the classification scheme implemented in <a href="http://planttfdb.gao-lab.org/" class="external-link">PlantTFDB</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <em><a href="https://github.com/almeidasilvaf/tfhunter" class="external-link">tfhunter</a></em> with
the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"tfhunter"</span><span class="op">)</span></span></code></pre></div>
<p>Loading package after installation:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/tfhunter" class="external-link">tfhunter</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>In this vignette, we will use protein sequences of TFs from the algae
species <em>Galdieria sulphuraria</em> as an example, as its proteome is
very small. The proteome file was downloaded from the PLAZA Diatoms
database <span class="citation">(Osuna-Cruz et al. 2020)</span>, and it
was filtered to keep only TFs for demonstration purposes. The object
<code>gsu</code> stores the protein sequences in an
<code>AAStringSet</code> object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span></span>
<span><span class="va">gsu</span></span>
<span><span class="co">#&gt; AAStringSet object of length 290:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   383 MSVLLQTSRGDIVVDLFTDLAP...EKALFQEHRRSSYSKTSKRYK* gsu00340.1</span></span>
<span><span class="co">#&gt;   [2]   301 MESMSQTKRWSCVIYQVDSILP...PNISLQQLLGDEDEISWSGLP* gsu04730.1</span></span>
<span><span class="co">#&gt;   [3]   232 MEKGPCSHKWKQLHVGDYQHDN...PNISLQQLLGDEDEISWSGLP* gsu05000.1</span></span>
<span><span class="co">#&gt;   [4]   340 MDVPSYRFECTPLVEVKKEARD...EVTQSSTRNCNDKEWNPNANE* gsu06140.1</span></span>
<span><span class="co">#&gt;   [5]   120 MAPKERSKPKSRSSKAGLQFPV...SGGVLPNVHPNLLPKKKAKEE* gsu06160.1</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [286]   394 MSKLEAASDSGSLKTSSCSFQE...NETNSHDKPGEDRMNIQENTS* gsu98790.1</span></span>
<span><span class="co">#&gt; [287]   365 MITEVDNPMSFVHSEHFMNYSS...RRHSRLPVFQTLEEKSDIHSK* gsu99250.1</span></span>
<span><span class="co">#&gt; [288]   365 MITEVDNPMSFVHSEHFMNYSS...RRHSRLPVFQTLEEKSDIHSK* gsu99270.1</span></span>
<span><span class="co">#&gt; [289]   284 MTSFYIKKGITFSSIVYNHNYK...LFQRIIEINKNYNPLIQLQRI* AIG92462.1</span></span>
<span><span class="co">#&gt; [290]   219 MKYKLLVIDDELSIRQSLKKYL...TFTRSRTELVRYAIKNNLIIE* AIG92471.1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="algorithm-description">Algorithm description<a class="anchor" aria-label="anchor" href="#algorithm-description"></a>
</h2>
<p>TF identification and classification is based on the presence of
signature protein domains, which are identified using profile hidden
Markov models (HMMs). The family classification scheme is the same as
the one used by PlantTFDB <span class="citation">(Jin et al.
2016)</span>, and it is summarized below: <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;strong&gt;Tip:&lt;/strong&gt; You can access this classification
scheme in your R session by loading the data frame
&lt;code&gt;data(classification_scheme)&lt;/code&gt;.&lt;/p&gt;"><sup>1</sup></a></p>
<table class="table">
<colgroup>
<col width="2%">
<col width="11%">
<col width="10%">
<col width="22%">
<col width="22%">
<col width="30%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Family</th>
<th align="left">Subfamily</th>
<th align="left">DBD</th>
<th align="left">Auxiliary</th>
<th align="left">Forbidden</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="left">AP2/ERF</td>
<td align="left">AP2</td>
<td align="left">AP2 (&gt;=2) (PF00847)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">AP2/ERF</td>
<td align="left">ERF</td>
<td align="left">AP2 (1) (PF00847)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">AP2/ERF</td>
<td align="left">RAV</td>
<td align="left">AP2 (PF00847) and B3 (PF02362)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">B3 superfamily</td>
<td align="left">ARF</td>
<td align="left">B3 (PF02362)</td>
<td align="left">Auxin_resp (PF06507)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">B3 superfamily</td>
<td align="left">B3</td>
<td align="left">B3 (PF02362)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">BBR-BPC</td>
<td align="left">BBR-BPC</td>
<td align="left">GAGA_bind (PF06217)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">BES1</td>
<td align="left">BES1</td>
<td align="left">DUF822 (PF05687)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">bHLH</td>
<td align="left">bHLH</td>
<td align="left">HLH (PF00010)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">bZIP</td>
<td align="left">bZIP</td>
<td align="left">bZIP_1 (PF00170)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">C2C2</td>
<td align="left">CO-like</td>
<td align="left">zf-B_box (PF00643)</td>
<td align="left">CCT (PF06203)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="left">C2C2</td>
<td align="left">Dof</td>
<td align="left">Zf-Dof (PF02701)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">13</td>
<td align="left">C2C2</td>
<td align="left">GATA</td>
<td align="left">GATA-zf (PF00320)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">14</td>
<td align="left">C2C2</td>
<td align="left">LSD</td>
<td align="left">Zf-LSD1 (PF06943)</td>
<td align="left">NA</td>
<td align="left">Peptidase_C14 (PF00656)</td>
</tr>
<tr class="even">
<td align="left">15</td>
<td align="left">C2C2</td>
<td align="left">YABBY</td>
<td align="left">YABBY (PF04690)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">16</td>
<td align="left">C2H2</td>
<td align="left">C2H2</td>
<td align="left">zf-C2H2 (PF00096)</td>
<td align="left">NA</td>
<td align="left">RNase_T (PF00929)</td>
</tr>
<tr class="even">
<td align="left">17</td>
<td align="left">C3H</td>
<td align="left">C3H</td>
<td align="left">Zf-CCCH (PF00642)</td>
<td align="left">NA</td>
<td align="left">RRM_1 (PF00076) or Helicase_C (PF00271)</td>
</tr>
<tr class="odd">
<td align="left">18</td>
<td align="left">CAMTA</td>
<td align="left">CAMTA</td>
<td align="left">CG1 (PF03859)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">19</td>
<td align="left">CPP</td>
<td align="left">CPP</td>
<td align="left">TCR (PF03638)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">20</td>
<td align="left">DBB</td>
<td align="left">DBB</td>
<td align="left">zf-B_box (&gt;=2) (PF00643)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">21</td>
<td align="left">E2F/DP</td>
<td align="left">E2F/DP</td>
<td align="left">E2F_TDP (PF02319)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">22</td>
<td align="left">EIL</td>
<td align="left">EIL</td>
<td align="left">EIN3 (PF04873)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">23</td>
<td align="left">FAR1</td>
<td align="left">FAR1</td>
<td align="left">FAR1 (PF03101)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">24</td>
<td align="left">GARP</td>
<td align="left">ARR-B</td>
<td align="left">G2-like</td>
<td align="left">Response_reg (PF00072)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">25</td>
<td align="left">GARP</td>
<td align="left">G2-like</td>
<td align="left">G2-like</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">26</td>
<td align="left">GeBP</td>
<td align="left">GeBP</td>
<td align="left">DUF573 (PF04504)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">27</td>
<td align="left">GRAS</td>
<td align="left">GRAS</td>
<td align="left">GRAS (PF03514)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">28</td>
<td align="left">GRF</td>
<td align="left">GRF</td>
<td align="left">WRC (PF08879)</td>
<td align="left">QLQ (PF08880)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">29</td>
<td align="left">HB</td>
<td align="left">HD-ZIP</td>
<td align="left">Homeobox (PF00046)</td>
<td align="left">HD-ZIP_I/II or SMART (PF01852)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">30</td>
<td align="left">HB</td>
<td align="left">TALE</td>
<td align="left">Homeobox (PF00046)</td>
<td align="left">BELL or ELK (PF03789)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">31</td>
<td align="left">HB</td>
<td align="left">WOX</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">Wus type homeobox</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">32</td>
<td align="left">HB</td>
<td align="left">HB-PHD</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">PHD (PF00628)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">33</td>
<td align="left">HB</td>
<td align="left">HB-other</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">34</td>
<td align="left">HRT-like</td>
<td align="left">HRT-like</td>
<td align="left">HRT-like</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">35</td>
<td align="left">HSF</td>
<td align="left">HSF</td>
<td align="left">HSF_dna_bind (PF00447)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">36</td>
<td align="left">LBD (AS2/LOB)</td>
<td align="left">LBD (AS2/LOB)</td>
<td align="left">DUF260 (PF03195)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">37</td>
<td align="left">LFY</td>
<td align="left">LFY</td>
<td align="left">FLO_LFY (PF01698)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">38</td>
<td align="left">MADS</td>
<td align="left">M_type</td>
<td align="left">SRF-TF (PF00319)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">39</td>
<td align="left">MADS</td>
<td align="left">MIKC</td>
<td align="left">SRF-TF (PF00319)</td>
<td align="left">K-box (PF01486)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">40</td>
<td align="left">MYB superfamily</td>
<td align="left">MYB</td>
<td align="left">Myb_dna_bind (&gt;=2) (PF00249)</td>
<td align="left">NA</td>
<td align="left">SWIRM (PF04433)</td>
</tr>
<tr class="even">
<td align="left">41</td>
<td align="left">MYB superfamily</td>
<td align="left">MYB_related</td>
<td align="left">Myb_dna_bind (1) (PF00249)</td>
<td align="left">NA</td>
<td align="left">SWIRM (PF04433)</td>
</tr>
<tr class="odd">
<td align="left">42</td>
<td align="left">NAC</td>
<td align="left">NAC</td>
<td align="left">NAM (PF02365)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">43</td>
<td align="left">NF-X1</td>
<td align="left">NF-X1</td>
<td align="left">Zf-NF-X1 (PF01422)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">44</td>
<td align="left">NF-Y</td>
<td align="left">NF-YA</td>
<td align="left">CBFB_NFYA (PF02045)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">45</td>
<td align="left">NF-Y</td>
<td align="left">NF-YB</td>
<td align="left">NF-YB</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">46</td>
<td align="left">NF-Y</td>
<td align="left">NF-YC</td>
<td align="left">NF-YC</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">47</td>
<td align="left">Nin-like</td>
<td align="left">Nin-like</td>
<td align="left">RWP-RK (PF02042)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">48</td>
<td align="left">NZZ/SPL</td>
<td align="left">NZZ/SPL</td>
<td align="left">NOZZLE (PF08744)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">49</td>
<td align="left">S1Fa-like</td>
<td align="left">S1Fa-like</td>
<td align="left">S1FA (PF04689)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">50</td>
<td align="left">SAP</td>
<td align="left">SAP</td>
<td align="left">SAP</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">51</td>
<td align="left">SBP</td>
<td align="left">SBP</td>
<td align="left">SBP (PF03110)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">52</td>
<td align="left">SRS</td>
<td align="left">SRS</td>
<td align="left">DUF702 (PF05142)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">53</td>
<td align="left">STAT</td>
<td align="left">STAT</td>
<td align="left">STAT</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">54</td>
<td align="left">TCP</td>
<td align="left">TCP</td>
<td align="left">TCP (PF03634)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">55</td>
<td align="left">Trihelix</td>
<td align="left">Trihelix</td>
<td align="left">Trihelix</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">56</td>
<td align="left">VOZ</td>
<td align="left">VOZ</td>
<td align="left">VOZ</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">57</td>
<td align="left">Whirly</td>
<td align="left">Whirly</td>
<td align="left">Whirly (PF08536)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">58</td>
<td align="left">WRKY</td>
<td align="left">WRKY</td>
<td align="left">WRKY (PF03106)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">59</td>
<td align="left">ZF-HD</td>
<td align="left">ZF-HD</td>
<td align="left">ZF-HD_dimer (PF04770)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="identifying-and-classifying-tfs">Identifying and classifying TFs<a class="anchor" aria-label="anchor" href="#identifying-and-classifying-tfs"></a>
</h2>
<p>To identify TFs from protein sequence data, you will use the function
<code><a href="../reference/annotate_pfam.html">annotate_pfam()</a></code>. This function takes as input an
<code>AAStringSet</code> object <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;strong&gt;Tip:&lt;/strong&gt; If you have protein sequences in a
FASTA file, you can read them into an &lt;code&gt;AAStringSet&lt;/code&gt; object
with the function &lt;code&gt;readAAStringSet()&lt;/code&gt; from the &lt;em&gt;&lt;a href="https://bioconductor.org/packages/3.15/Biostrings" class="external-link"&gt;Biostrings&lt;/a&gt;&lt;/em&gt;
package.&lt;/p&gt;'><sup>2</sup></a> and returns a data frame of protein domains
associated with each sequence. The HMMER program <span class="citation">(Finn, Clements, and Eddy 2011)</span> is used to scan
protein sequences for the presence of DNA-binding protein domains, as
well as auxiliary and forbidden domains. Pre-built HMM profiles can be
found in the <em>extdata/</em> directory of this package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Annotate TF-related domains using a local installation of HMMER</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/hmmer_is_installed.html">hmmer_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">gsu_annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_pfam.html">annotate_pfam</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span></span>
<span><span class="op">}</span> </span>
<span></span>
<span><span class="co"># Take a look at the first few lines of the output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span><span class="co">#&gt;          Gene  Domain</span></span>
<span><span class="co">#&gt; 1 gsu144370.1 PF00010</span></span>
<span><span class="co">#&gt; 2 gsu140730.1 PF00010</span></span>
<span><span class="co">#&gt; 3 gsu127100.1 PF00046</span></span>
<span><span class="co">#&gt; 4 gsu109490.1 PF00046</span></span>
<span><span class="co">#&gt; 5  AIG92462.1 PF00072</span></span>
<span><span class="co">#&gt; 6  AIG92471.1 PF00072</span></span></code></pre></div>
<p>Now that we have our TF-related domains, we can classify TFs in
families with the function <code><a href="../reference/classify_tfs.html">classify_tfs()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu_families</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Classify TFs into families</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/hmmer_is_installed.html">hmmer_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">gsu_families</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classify_tfs.html">classify_tfs</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span><span class="op">}</span> </span>
<span></span>
<span><span class="co"># Take a look at the output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gsu_families</span><span class="op">)</span></span>
<span><span class="co">#&gt;          Gene      Family</span></span>
<span><span class="co">#&gt; 1  gsu04730.1    Nin-like</span></span>
<span><span class="co">#&gt; 2  gsu05000.1    Nin-like</span></span>
<span><span class="co">#&gt; 3  gsu06140.1     G2-like</span></span>
<span><span class="co">#&gt; 4  gsu06140.1 MYB-related</span></span>
<span><span class="co">#&gt; 5  gsu09770.1         C3H</span></span>
<span><span class="co">#&gt; 6 gsu100410.1    Nin-like</span></span>
<span></span>
<span><span class="co"># Count number of TFs per family</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">gsu_families</span><span class="op">$</span><span class="va">Family</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;        bHLH        bZIP        C2H2         C3H         CPP      E2F/DP </span></span>
<span><span class="co">#&gt;           2           8           4          13           3           8 </span></span>
<span><span class="co">#&gt;     G2-like        GATA    HB-other         HSF         LSD      M-type </span></span>
<span><span class="co">#&gt;           2           8           2           6           3           1 </span></span>
<span><span class="co">#&gt;         MYB MYB-related       NF-X1       NF-YA       NF-YB       NF-YC </span></span>
<span><span class="co">#&gt;          16          28           1           1           4           5 </span></span>
<span><span class="co">#&gt;    Nin-like </span></span>
<span><span class="co">#&gt;          10</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="counting-tfs-per-family-in-multiple-species-at-once">Counting TFs per family in multiple species at once<a class="anchor" aria-label="anchor" href="#counting-tfs-per-family-in-multiple-species-at-once"></a>
</h2>
<p>If you want to get TF counts per family for multiple species, you can
use the function <code><a href="../reference/get_tf_counts.html">get_tf_counts()</a></code>. This function takes a
list of <code>AAStringSet</code> objects containing proteomes as input,
and it returns a <code>SummarizedExperiment</code> object containing TF
counts per family in each species, as well as species metadata
(optional). If you are not familiar with the
<code>SummarizedExperiment</code> class, you should consider checking
the vignettes of the <em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
Bioconductor package.</p>
<p>To demonstrate how <code><a href="../reference/get_tf_counts.html">get_tf_counts()</a></code> works, we will
simulate a list of 4 <code>AAStringSet</code> objects by sampling 100
random genes from the example data set <code>gsu</code> 4 times. We will
also create a toy species metadata data frame for each “species”
(simulated).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></span>
<span></span>
<span><span class="co"># Simulate 4 different species by sampling 100 random genes from `gsu`</span></span>
<span><span class="va">proteomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Gsu1 <span class="op">=</span> <span class="va">gsu</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    Gsu2 <span class="op">=</span> <span class="va">gsu</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    Gsu3 <span class="op">=</span> <span class="va">gsu</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span>,</span>
<span>    Gsu4 <span class="op">=</span> <span class="va">gsu</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span><span class="va">proteomes</span></span>
<span><span class="co">#&gt; $Gsu1</span></span>
<span><span class="co">#&gt; AAStringSet object of length 50:</span></span>
<span><span class="co">#&gt;      width seq                                              names               </span></span>
<span><span class="co">#&gt;  [1]   405 MSQDTYTFEKLGVCKILCEECKN...SSSYNCPTDERMESEEEKLET* gsu49120.1</span></span>
<span><span class="co">#&gt;  [2]   292 MGRSTTVFVGNIAYNTSEEQLQE...TLGAQMGQPGLGSSAFSSNNN* gsu102250.1</span></span>
<span><span class="co">#&gt;  [3]   530 MTFTRIENNLKKYFGYENFRPLQ...KMSKTQNINQKTLLRWFSETM* gsu56420.1</span></span>
<span><span class="co">#&gt;  [4]   672 MNRSLFTPTLRWSQWFAKTCTTL...EKRKLSVVHNKTQQIVSFNRH* gsu22760.1</span></span>
<span><span class="co">#&gt;  [5]   606 MSQERLTPFERHLKKVEEKNQRE...EEDLDGVPLEEDEEAIEIEIK* gsu68790.1</span></span>
<span><span class="co">#&gt;  ...   ... ...</span></span>
<span><span class="co">#&gt; [46]   465 MNMFSYSTIETPKYVRSNSSDDR...STQMSHLLSVALQDWVEWNQE* gsu47220.1</span></span>
<span><span class="co">#&gt; [47]  2225 MSFPPKFRNYLLAQKGQPVSVEQ...MWRWVDNKSQFLSSYQVAWNE* gsu96990.1</span></span>
<span><span class="co">#&gt; [48]   292 MRGKSKSVVFPASRIKRIMRINE...LDKVESPSRVFISIEELVHSV* gsu124540.1</span></span>
<span><span class="co">#&gt; [49]   477 MDSSKKSTNPKLSESGTKDNRGN...PSDTTAPQVAVNVHAGNGSSK* gsu67550.1</span></span>
<span><span class="co">#&gt; [50]   740 MERLQPPYRYLIILDLEATAVDL...CTQDKSVSLGSVSLEGKGDSV* gsu102410.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Gsu2</span></span>
<span><span class="co">#&gt; AAStringSet object of length 50:</span></span>
<span><span class="co">#&gt;      width seq                                              names               </span></span>
<span><span class="co">#&gt;  [1]   764 MSQSVPVKNDTEDSCGVQKLSNA...NSDDTSRIRRRTLNVHDLLSE* gsu21860.1</span></span>
<span><span class="co">#&gt;  [2]   406 MQPHVSDHRYPTTVEQREYHSGS...QTTDVTGGAVFLRKETEHKDI* gsu140730.1</span></span>
<span><span class="co">#&gt;  [3]   714 MEDERNSRLLIQGLPKYIAEKRL...LVFDYLVNWMALIVALVLSSF* gsu84990.1</span></span>
<span><span class="co">#&gt;  [4]   236 MDTSEQGSEQGEESISNNSQQLC...YTNTASHRFRKLLKASVGDTS* gsu72770.1</span></span>
<span><span class="co">#&gt;  [5]   524 MKKNAKEIGAQYSALVFIHVALF...PVSTVPYFVMDNNSGGSYSFV* gsu136350.1</span></span>
<span><span class="co">#&gt;  ...   ... ...</span></span>
<span><span class="co">#&gt; [46]   284 MTSFYIKKGITFSSIVYNHNYKF...LFQRIIEINKNYNPLIQLQRI* AIG92462.1</span></span>
<span><span class="co">#&gt; [47]   546 MAPRLNKTTQTKLKKQSSFREQP...NACGLFWAKHKQLRPKEKWVR* gsu134260.1</span></span>
<span><span class="co">#&gt; [48]   310 MSGGSGIYQPSGMSLYVGNLDPR...EIAGCVVQCEWGREGLKSRYF* gsu10640.1</span></span>
<span><span class="co">#&gt; [49]   445 MEPISKDDVYGGFSTVENCDSSM...NCKCVDCKNQSSLSLLKNTAM* gsu21090.1</span></span>
<span><span class="co">#&gt; [50]   318 MFICGHMIQNLVSVCAHSRIFCI...DCSYIFATDASDYPPPWQYFP* gsu64800.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Gsu3</span></span>
<span><span class="co">#&gt; AAStringSet object of length 50:</span></span>
<span><span class="co">#&gt;      width seq                                              names               </span></span>
<span><span class="co">#&gt;  [1]   196 MAYLYEDRPVTLYRDRRFQGTQE...RDETDEVFQTEKNPRFREEED* gsu18660.1</span></span>
<span><span class="co">#&gt;  [2]   171 MEDRMQVVVSETSKGEERGRGRG...SWAFSRGPLGTRLSTRRRSEK* gsu34810.1</span></span>
<span><span class="co">#&gt;  [3]   461 MTERIDKSRRKKYVLTKKREYWT...HKESFSKRTYPDSVQAVIVGQ* gsu38570.1</span></span>
<span><span class="co">#&gt;  [4]  1207 MELVSGPLLDQFPFVAGHSRQTL...KYGREHHWQYSLEHPFVSPIL* gsu143710.1</span></span>
<span><span class="co">#&gt;  [5]   928 MFILVVEVKVEEYFSFQLDDFQQ...LRRVLDILRQIPRLPAKQWLS* gsu79190.1</span></span>
<span><span class="co">#&gt;  ...   ... ...</span></span>
<span><span class="co">#&gt; [46]   150 MAAVEDNRVFVGGLPWSVGEDDL...GHGHGGRGGRSGGFRRREDFE* gsu58080.1</span></span>
<span><span class="co">#&gt; [47]   122 MAPKERSKPKSRSSKAGLQFPVG...GVLPNVHPNLLPKKKAKEDMQ* gsu59350.1</span></span>
<span><span class="co">#&gt; [48]   680 MWSTVDYSLNCDEEFRELSNAAA...QQSTNVVYPSNTNNSETCENS* gsu44800.1</span></span>
<span><span class="co">#&gt; [49]   148 MVANGEPGVIYIGHLPHGFYENE...ERRNKELEVKLKKLGVSFSLS* gsu97960.1</span></span>
<span><span class="co">#&gt; [50]   226 MAYRKLETRVPSYLDEVLGKVSS...SAPEGVWRCPDCRSGGANRAR* gsu111770.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Gsu4</span></span>
<span><span class="co">#&gt; AAStringSet object of length 50:</span></span>
<span><span class="co">#&gt;      width seq                                              names               </span></span>
<span><span class="co">#&gt;  [1]   740 MERLQPPYRYLIILDLEATAVDL...CTQDKSVSLGSVSLEGKGDSV* gsu102410.1</span></span>
<span><span class="co">#&gt;  [2]   701 MFKSSLLTFPALKTVVGAQDQYT...NLTKSFDLKKSQLSKRKKKWK* gsu110850.1</span></span>
<span><span class="co">#&gt;  [3]   484 MSEVSWEAGRSPVQPGKDHKSSS...DRSSTENRNSGRRRSVEGRAR* gsu114840.1</span></span>
<span><span class="co">#&gt;  [4]   483 MEEERKQKKGTGTSVSKTRAVQE...AKSADDESSRPVRQYDVENVA* gsu100590.1</span></span>
<span><span class="co">#&gt;  [5]   122 MAPKERSKPKSRSSKAGLQFPVG...GVLPNVHPNLLPKKKAKEDMQ* gsu59350.1</span></span>
<span><span class="co">#&gt;  ...   ... ...</span></span>
<span><span class="co">#&gt; [46]   798 MVLYQSYSSDTNSDVKPSEVSNS...EDENNKILCLCGAPTCRKFLN* gsu40400.1</span></span>
<span><span class="co">#&gt; [47]   383 MSVLLQTSRGDIVVDLFTDLAPL...EKALFQEHRRSSYSKTSKRYK* gsu139150.1</span></span>
<span><span class="co">#&gt; [48]   518 MRSGTTLSSLHNSHTEDATSLRA...EIGDIASLLEGEEVNYERLER* gsu32730.1</span></span>
<span><span class="co">#&gt; [49]   287 MKASQVLASQLCELCQSANSSIY...RKQLAERRCRFKGRFIKNTAS* gsu55840.1</span></span>
<span><span class="co">#&gt; [50]  1886 MDDTEYVPVKKRRQRILQQAKEL...TEVTRKQLIYYLSDVLANNKE* gsu43660.1</span></span></code></pre></div>
<p>Great, now we have a list of 4 <code>AAStringSet</code> objects.
Let’s also create a simulated species metadata data frame for each
“species” (simulated).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create simulated species metadata</span></span>
<span><span class="va">species_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">proteomes</span><span class="op">)</span>,</span>
<span>    Division <span class="op">=</span> <span class="st">"Rhodophyta"</span>,</span>
<span>    Origin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"US"</span>, <span class="st">"Belgium"</span>, <span class="st">"China"</span>, <span class="st">"Brazil"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">species_metadata</span></span>
<span><span class="co">#&gt;        Division  Origin</span></span>
<span><span class="co">#&gt; Gsu1 Rhodophyta      US</span></span>
<span><span class="co">#&gt; Gsu2 Rhodophyta Belgium</span></span>
<span><span class="co">#&gt; Gsu3 Rhodophyta   China</span></span>
<span><span class="co">#&gt; Gsu4 Rhodophyta  Brazil</span></span></code></pre></div>
<p>You can add as many columns as you want to the species metadata data
frame, but make sure that <strong>species names are in row
names</strong>, and that <code>names(proteomes)</code> match
<code>rownames(species)</code>, otherwise <code><a href="../reference/get_tf_counts.html">get_tf_counts()</a></code>
will return an error.</p>
<p>Now that we have a list of <code>AAStringSet</code> objects and
species metadata, we can execute <code><a href="../reference/get_tf_counts.html">get_tf_counts()</a></code> as
below:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tf_counts</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get TF counts per family in each species as a SummarizedExperiment object</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/hmmer_is_installed.html">hmmer_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">tf_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_tf_counts.html">get_tf_counts</a></span><span class="op">(</span><span class="va">proteomes</span>, <span class="va">species_metadata</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Take a look at the SummarizedExperiment object</span></span>
<span><span class="va">tf_counts</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 19 4 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(19): C2H2 C3H ... NF-YA NF-YB</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(4): Gsu1 Gsu2 Gsu3 Gsu4</span></span>
<span><span class="co">#&gt; colData names(2): Division Origin</span></span>
<span></span>
<span><span class="co"># Look at the matrix of counts: assay() function from SummarizedExperiment</span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">tf_counts</span><span class="op">)</span></span>
<span><span class="co">#&gt;             Gsu1 Gsu2 Gsu3 Gsu4</span></span>
<span><span class="co">#&gt; C2H2           3    1    1    2</span></span>
<span><span class="co">#&gt; C3H            3    2    0    3</span></span>
<span><span class="co">#&gt; CPP            1    1    1    0</span></span>
<span><span class="co">#&gt; E2F/DP         1    2    2    2</span></span>
<span><span class="co">#&gt; GATA           1    2    1    0</span></span>
<span><span class="co">#&gt; HSF            2    0    0    3</span></span>
<span><span class="co">#&gt; MYB            3    3    0    2</span></span>
<span><span class="co">#&gt; MYB-related    4    5    3    9</span></span>
<span><span class="co">#&gt; NF-X1          1    0    0    0</span></span>
<span><span class="co">#&gt; NF-YC          3    0    0    1</span></span>
<span><span class="co">#&gt; Nin-like       2    0    1    1</span></span>
<span><span class="co">#&gt; bHLH           0    1    1    0</span></span>
<span><span class="co">#&gt; bZIP           0    2    4    0</span></span>
<span><span class="co">#&gt; G2-like        0    1    0    0</span></span>
<span><span class="co">#&gt; HB-other       0    1    0    0</span></span>
<span><span class="co">#&gt; LSD            0    1    0    0</span></span>
<span><span class="co">#&gt; M-type         0    1    1    0</span></span>
<span><span class="co">#&gt; NF-YA          0    1    0    0</span></span>
<span><span class="co">#&gt; NF-YB          0    2    0    1</span></span>
<span></span>
<span><span class="co"># Look at the species metadata: colData() function from SummarizedExperiment</span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tf_counts</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 4 rows and 2 columns</span></span>
<span><span class="co">#&gt;         Division      Origin</span></span>
<span><span class="co">#&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; Gsu1  Rhodophyta          US</span></span>
<span><span class="co">#&gt; Gsu2  Rhodophyta     Belgium</span></span>
<span><span class="co">#&gt; Gsu3  Rhodophyta       China</span></span>
<span><span class="co">#&gt; Gsu4  Rhodophyta      Brazil</span></span></code></pre></div>
<p>Cool, huh? In real-world analyses, if you have TF counts per family
in multiple species, you can try to find associations between TF counts
and eco-evolutionary aspects of each species (e.g., higher frequencies
of a stress-related TF family in a species that inhabits a stressful
environment).</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2022-10-06</span></span>
<span><span class="co">#&gt;  pandoc   2.18 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  Biobase                2.56.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics           0.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.18  <span style="color: #555555;">2022-05-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.24.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings             2.64.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.29     <span style="color: #555555;">2022-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.4.0    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.4.1    <span style="color: #555555;">2022-09-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.2    <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.22.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.2    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.29   <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate               0.16     <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.32.4   <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.8    <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges          1.48.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  highr                  0.9      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.3    <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  IRanges                2.30.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.2    <span style="color: #555555;">2022-10-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.40     <span style="color: #555555;">2022-08-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.20-45  <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.5-1    <span style="color: #555555;">2022-09-13</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.8.1    <span style="color: #555555;">2022-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats            0.62.0   <span style="color: #555555;">2022-04-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.0.6    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr                  0.3.4    <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.2.3    <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.9 <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.0.6    <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.16     <span style="color: #555555;">2022-08-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot              2.0.3    <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors              0.34.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.2    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi                1.7.8    <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr                1.4.1    <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.26.1   <span style="color: #555555;">2022-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tfhunter             * 0.99.0   <span style="color: #555555;">2022-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  xfun                   0.33     <span style="color: #555555;">2022-09-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.36.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.5    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-finn2011hmmer" class="csl-entry">
Finn, Robert D, Jody Clements, and Sean R Eddy. 2011. <span>“HMMER Web
Server: Interactive Sequence Similarity Searching.”</span> <em>Nucleic
Acids Research</em> 39 (suppl_2): W29–37.
</div>
<div id="ref-jin2016planttfdb" class="csl-entry">
Jin, Jinpu, Feng Tian, De-Chang Yang, Yu-Qi Meng, Lei Kong, Jingchu Luo,
and Ge Gao. 2016. <span>“PlantTFDB 4.0: Toward a Central Hub for
Transcription Factors and Regulatory Interactions in Plants.”</span>
<em>Nucleic Acids Research</em>, gkw982.
</div>
<div id="ref-osuna2020seminavis" class="csl-entry">
Osuna-Cruz, Cristina Maria, Gust Bilcke, Emmelien Vancaester, Sam De
Decker, Atle M Bones, Per Winge, Nicole Poulsen, et al. 2020. <span>“The
Seminavis Robusta Genome Provides Insights into the Evolutionary
Adaptations of Benthic Diatoms.”</span> <em>Nature Communications</em>
11 (1): 1–13.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
