<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tfhunter">
<title>Genome-wide identification and classification of transcription factors in plant genomes • tfhunter</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.2/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Genome-wide identification and classification of transcription factors in plant genomes">
<meta property="og:description" content="tfhunter">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tfhunter</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_tfhunter.html">Genome-wide identification and classification of transcription factors in plant genomes</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/tfhunter/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genome-wide identification and classification of transcription factors in plant genomes</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent,
BelgiumDepartment of Plant Biotechnology and Bioinformatics, Ghent
University, Ghent,
Belgium<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent,
BelgiumDepartment of Plant Biotechnology and Bioinformatics, Ghent
University, Ghent, BelgiumCollege of Horticulture, Academy for Advanced
Interdisciplinary Studies, Nanjing Agricultural University, Nanjing,
ChinaCenter for Microbial Ecology and Genomics, Department of
Biochemistry, Genetics and Microbiology, University of Pretoria,
Pretoria, South Africa<br><h4 data-toc-skip class="date">2022-08-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/tfhunter/blob/HEAD/vignettes/vignette_tfhunter.Rmd" class="external-link"><code>vignettes/vignette_tfhunter.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_tfhunter.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Transcription factors (TFs) are proteins that bind to cis-regulatory
elements in promoter regions of genes and regulate their expression.
Identifying them in a genome is useful for a variety of reasons, such as
exploring their evolutionary history across clades and inferring gene
regulatory networks. <code>tfhunter</code> allows users to identify
plant TFs from whole-genome protein sequences and classify them into
families and subfamilies (when applicable) using the classification
scheme implemented in <a href="http://planttfdb.gao-lab.org/" class="external-link">PlantTFDB</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>tfhunter</code> with the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"tfhunter"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/tfhunter" class="external-link">tfhunter</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>In this vignette, we will use protein sequences of TFs from the algae
species <em>Galdieria sulphuraria</em> as an example. The proteome file
was downloaded from the PLAZA Diatoms database <span class="citation">(Osuna-Cruz et al. 2020)</span>, and it was filtered to
keep only TFs for demonstration purposes. The object <code>gsu</code>
stores the protein sequences in an AAStringSet object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu</span><span class="op">)</span></span>
<span><span class="va">gsu</span></span>
<span><span class="co">#&gt; AAStringSet object of length 290:</span></span>
<span><span class="co">#&gt;       width seq                                             names               </span></span>
<span><span class="co">#&gt;   [1]   383 MSVLLQTSRGDIVVDLFTDLAP...EKALFQEHRRSSYSKTSKRYK* gsu00340.1</span></span>
<span><span class="co">#&gt;   [2]   301 MESMSQTKRWSCVIYQVDSILP...PNISLQQLLGDEDEISWSGLP* gsu04730.1</span></span>
<span><span class="co">#&gt;   [3]   232 MEKGPCSHKWKQLHVGDYQHDN...PNISLQQLLGDEDEISWSGLP* gsu05000.1</span></span>
<span><span class="co">#&gt;   [4]   340 MDVPSYRFECTPLVEVKKEARD...EVTQSSTRNCNDKEWNPNANE* gsu06140.1</span></span>
<span><span class="co">#&gt;   [5]   120 MAPKERSKPKSRSSKAGLQFPV...SGGVLPNVHPNLLPKKKAKEE* gsu06160.1</span></span>
<span><span class="co">#&gt;   ...   ... ...</span></span>
<span><span class="co">#&gt; [286]   394 MSKLEAASDSGSLKTSSCSFQE...NETNSHDKPGEDRMNIQENTS* gsu98790.1</span></span>
<span><span class="co">#&gt; [287]   365 MITEVDNPMSFVHSEHFMNYSS...RRHSRLPVFQTLEEKSDIHSK* gsu99250.1</span></span>
<span><span class="co">#&gt; [288]   365 MITEVDNPMSFVHSEHFMNYSS...RRHSRLPVFQTLEEKSDIHSK* gsu99270.1</span></span>
<span><span class="co">#&gt; [289]   284 MTSFYIKKGITFSSIVYNHNYK...LFQRIIEINKNYNPLIQLQRI* AIG92462.1</span></span>
<span><span class="co">#&gt; [290]   219 MKYKLLVIDDELSIRQSLKKYL...TFTRSRTELVRYAIKNNLIIE* AIG92471.1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="algorithm-description">Algorithm description<a class="anchor" aria-label="anchor" href="#algorithm-description"></a>
</h2>
<p>TF identification and classification is based on the presence of
signature protein domains, which are identified using profile hidden
Markov models (HMMs). The family classification scheme is the same as
the one used by PlantTFDB <span class="citation">(Jin et al.
2016)</span>, and it is summarized below: <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;strong&gt;Tip:&lt;/strong&gt; You can access this classification
scheme in your R session by loading the data frame
&lt;code&gt;data(classification_scheme)&lt;/code&gt;.&lt;/p&gt;"><sup>1</sup></a></p>
<table class="table">
<colgroup>
<col width="2%">
<col width="11%">
<col width="10%">
<col width="22%">
<col width="22%">
<col width="30%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Family</th>
<th align="left">Subfamily</th>
<th align="left">DBD</th>
<th align="left">Auxiliary</th>
<th align="left">Forbidden</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="left">AP2/ERF</td>
<td align="left">AP2</td>
<td align="left">AP2 (&gt;=2) (PF00847)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">AP2/ERF</td>
<td align="left">ERF</td>
<td align="left">AP2 (1) (PF00847)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">AP2/ERF</td>
<td align="left">RAV</td>
<td align="left">AP2 (PF00847) and B3 (PF02362)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">B3 superfamily</td>
<td align="left">ARF</td>
<td align="left">B3 (PF02362)</td>
<td align="left">Auxin_resp (PF06507)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">B3 superfamily</td>
<td align="left">B3</td>
<td align="left">B3 (PF02362)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">BBR-BPC</td>
<td align="left">BBR-BPC</td>
<td align="left">GAGA_bind (PF06217)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">BES1</td>
<td align="left">BES1</td>
<td align="left">DUF822 (PF05687)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">bHLH</td>
<td align="left">bHLH</td>
<td align="left">HLH (PF00010)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">bZIP</td>
<td align="left">bZIP</td>
<td align="left">bZIP_1 (PF00170)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">C2C2</td>
<td align="left">CO-like</td>
<td align="left">zf-B_box (PF00643)</td>
<td align="left">CCT (PF06203)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="left">C2C2</td>
<td align="left">Dof</td>
<td align="left">Zf-Dof (PF02701)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">13</td>
<td align="left">C2C2</td>
<td align="left">GATA</td>
<td align="left">GATA-zf (PF00320)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">14</td>
<td align="left">C2C2</td>
<td align="left">LSD</td>
<td align="left">Zf-LSD1 (PF06943)</td>
<td align="left">NA</td>
<td align="left">Peptidase_C14 (PF00656)</td>
</tr>
<tr class="even">
<td align="left">15</td>
<td align="left">C2C2</td>
<td align="left">YABBY</td>
<td align="left">YABBY (PF04690)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">16</td>
<td align="left">C2H2</td>
<td align="left">C2H2</td>
<td align="left">zf-C2H2 (PF00096)</td>
<td align="left">NA</td>
<td align="left">RNase_T (PF00929)</td>
</tr>
<tr class="even">
<td align="left">17</td>
<td align="left">C3H</td>
<td align="left">C3H</td>
<td align="left">Zf-CCCH (PF00642)</td>
<td align="left">NA</td>
<td align="left">RRM_1 (PF00076) or Helicase_C (PF00271)</td>
</tr>
<tr class="odd">
<td align="left">18</td>
<td align="left">CAMTA</td>
<td align="left">CAMTA</td>
<td align="left">CG1 (PF03859)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">19</td>
<td align="left">CPP</td>
<td align="left">CPP</td>
<td align="left">TCR (PF03638)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">20</td>
<td align="left">DBB</td>
<td align="left">DBB</td>
<td align="left">zf-B_box (&gt;=2) (PF00643)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">21</td>
<td align="left">E2F/DP</td>
<td align="left">E2F/DP</td>
<td align="left">E2F_TDP (PF02319)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">22</td>
<td align="left">EIL</td>
<td align="left">EIL</td>
<td align="left">EIN3 (PF04873)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">23</td>
<td align="left">FAR1</td>
<td align="left">FAR1</td>
<td align="left">FAR1 (PF03101)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">24</td>
<td align="left">GARP</td>
<td align="left">ARR-B</td>
<td align="left">G2-like</td>
<td align="left">Response_reg (PF00072)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">25</td>
<td align="left">GARP</td>
<td align="left">G2-like</td>
<td align="left">G2-like</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">26</td>
<td align="left">GeBP</td>
<td align="left">GeBP</td>
<td align="left">DUF573 (PF04504)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">27</td>
<td align="left">GRAS</td>
<td align="left">GRAS</td>
<td align="left">GRAS (PF03514)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">28</td>
<td align="left">GRF</td>
<td align="left">GRF</td>
<td align="left">WRC (PF08879)</td>
<td align="left">QLQ (PF08880)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">29</td>
<td align="left">HB</td>
<td align="left">HD-ZIP</td>
<td align="left">Homeobox (PF00046)</td>
<td align="left">HD-ZIP_I/II or SMART (PF01852)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">30</td>
<td align="left">HB</td>
<td align="left">TALE</td>
<td align="left">Homeobox (PF00046)</td>
<td align="left">BELL or ELK (PF03789)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">31</td>
<td align="left">HB</td>
<td align="left">WOX</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">Wus type homeobox</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">32</td>
<td align="left">HB</td>
<td align="left">HB-PHD</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">PHD (PF00628)</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">33</td>
<td align="left">HB</td>
<td align="left">HB-other</td>
<td align="left">homeobox (PF00046)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">34</td>
<td align="left">HRT-like</td>
<td align="left">HRT-like</td>
<td align="left">HRT-like</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">35</td>
<td align="left">HSF</td>
<td align="left">HSF</td>
<td align="left">HSF_dna_bind (PF00447)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">36</td>
<td align="left">LBD (AS2/LOB)</td>
<td align="left">LBD (AS2/LOB)</td>
<td align="left">DUF260 (PF03195)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">37</td>
<td align="left">LFY</td>
<td align="left">LFY</td>
<td align="left">FLO_LFY (PF01698)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">38</td>
<td align="left">MADS</td>
<td align="left">M_type</td>
<td align="left">SRF-TF (PF00319)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">39</td>
<td align="left">MADS</td>
<td align="left">MIKC</td>
<td align="left">SRF-TF (PF00319)</td>
<td align="left">K-box (PF01486)</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">40</td>
<td align="left">MYB superfamily</td>
<td align="left">MYB</td>
<td align="left">Myb_dna_bind (&gt;=2) (PF00249)</td>
<td align="left">NA</td>
<td align="left">SWIRM (PF04433)</td>
</tr>
<tr class="even">
<td align="left">41</td>
<td align="left">MYB superfamily</td>
<td align="left">MYB_related</td>
<td align="left">Myb_dna_bind (1) (PF00249)</td>
<td align="left">NA</td>
<td align="left">SWIRM (PF04433)</td>
</tr>
<tr class="odd">
<td align="left">42</td>
<td align="left">NAC</td>
<td align="left">NAC</td>
<td align="left">NAM (PF02365)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">43</td>
<td align="left">NF-X1</td>
<td align="left">NF-X1</td>
<td align="left">Zf-NF-X1 (PF01422)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">44</td>
<td align="left">NF-Y</td>
<td align="left">NF-YA</td>
<td align="left">CBFB_NFYA (PF02045)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">45</td>
<td align="left">NF-Y</td>
<td align="left">NF-YB</td>
<td align="left">NF-YB</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">46</td>
<td align="left">NF-Y</td>
<td align="left">NF-YC</td>
<td align="left">NF-YC</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">47</td>
<td align="left">Nin-like</td>
<td align="left">Nin-like</td>
<td align="left">RWP-RK (PF02042)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">48</td>
<td align="left">NZZ/SPL</td>
<td align="left">NZZ/SPL</td>
<td align="left">NOZZLE (PF08744)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">49</td>
<td align="left">S1Fa-like</td>
<td align="left">S1Fa-like</td>
<td align="left">S1FA (PF04689)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">50</td>
<td align="left">SAP</td>
<td align="left">SAP</td>
<td align="left">SAP</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">51</td>
<td align="left">SBP</td>
<td align="left">SBP</td>
<td align="left">SBP (PF03110)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">52</td>
<td align="left">SRS</td>
<td align="left">SRS</td>
<td align="left">DUF702 (PF05142)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">53</td>
<td align="left">STAT</td>
<td align="left">STAT</td>
<td align="left">STAT</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">54</td>
<td align="left">TCP</td>
<td align="left">TCP</td>
<td align="left">TCP (PF03634)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">55</td>
<td align="left">Trihelix</td>
<td align="left">Trihelix</td>
<td align="left">Trihelix</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">56</td>
<td align="left">VOZ</td>
<td align="left">VOZ</td>
<td align="left">VOZ</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">57</td>
<td align="left">Whirly</td>
<td align="left">Whirly</td>
<td align="left">Whirly (PF08536)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">58</td>
<td align="left">WRKY</td>
<td align="left">WRKY</td>
<td align="left">WRKY (PF03106)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">59</td>
<td align="left">ZF-HD</td>
<td align="left">ZF-HD</td>
<td align="left">ZF-HD_dimer (PF04770)</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="identifying-and-classifying-tfs">Identifying and classifying TFs<a class="anchor" aria-label="anchor" href="#identifying-and-classifying-tfs"></a>
</h2>
<p>To identify TFs from protein sequence data, you will use the function
<code><a href="../reference/annotate_pfam.html">annotate_pfam()</a></code>. This function takes as input an
AAStringSet object <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;strong&gt;Tip:&lt;/strong&gt; If you have protein sequences in a
FASTA file, you can read them into an AAStringSet object with the
function &lt;code&gt;readAAStringSet()&lt;/code&gt; from the Biostrings package.&lt;/p&gt;"><sup>2</sup></a> and returns a data frame of protein domains
associated with each sequence. The HMMER program <span class="citation">(Finn, Clements, and Eddy 2011)</span> is used to scan
protein sequences for the presence of TF-related protein domains, and it
can be run either through the web server (by setting
<code>mode = web</code>) or through a local HMMER installation
(<code>mode = "local"</code>). We advise you to install HMMER in your
machine and set <code>mode = "local"</code>, as domain annotation with
the web server can take several minutes, especially for large
proteomes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Annotate TF-related domains using a local installation of HMMER</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/hmmer_is_installed.html">hmmer_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">gsu_annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotate_pfam.html">annotate_pfam</a></span><span class="op">(</span><span class="va">gsu</span>, mode <span class="op">=</span> <span class="st">"local"</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span><span class="co">#&gt;          Gene  Domain</span></span>
<span><span class="co">#&gt; 1 gsu144370.1 PF00010</span></span>
<span><span class="co">#&gt; 2 gsu140730.1 PF00010</span></span>
<span><span class="co">#&gt; 3 gsu127100.1 PF00046</span></span>
<span><span class="co">#&gt; 4 gsu109490.1 PF00046</span></span>
<span><span class="co">#&gt; 5  AIG92462.1 PF00072</span></span>
<span><span class="co">#&gt; 6  AIG92471.1 PF00072</span></span></code></pre></div>
<p>Now that we have our TF-related domains, we can classify TFs in
families with the function <code><a href="../reference/classify_tfs.html">classify_tfs()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/hmmer_is_installed.html">hmmer_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">gsu_families</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/classify_tfs.html">classify_tfs</a></span><span class="op">(</span><span class="va">gsu_annotation</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gsu_families</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gsu_families</span><span class="op">)</span></span>
<span><span class="co">#&gt;          Gene      Family</span></span>
<span><span class="co">#&gt; 1  gsu04730.1    Nin-like</span></span>
<span><span class="co">#&gt; 2  gsu05000.1    Nin-like</span></span>
<span><span class="co">#&gt; 3  gsu06140.1     G2-like</span></span>
<span><span class="co">#&gt; 4  gsu06140.1 MYB-related</span></span>
<span><span class="co">#&gt; 5  gsu09770.1         C3H</span></span>
<span><span class="co">#&gt; 6 gsu100410.1    Nin-like</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-finn2011hmmer" class="csl-entry">
Finn, Robert D, Jody Clements, and Sean R Eddy. 2011. <span>“HMMER Web
Server: Interactive Sequence Similarity Searching.”</span> <em>Nucleic
Acids Research</em> 39 (suppl_2): W29–37.
</div>
<div id="ref-jin2016planttfdb" class="csl-entry">
Jin, Jinpu, Feng Tian, De-Chang Yang, Yu-Qi Meng, Lei Kong, Jingchu Luo,
and Ge Gao. 2016. <span>“PlantTFDB 4.0: Toward a Central Hub for
Transcription Factors and Regulatory Interactions in Plants.”</span>
<em>Nucleic Acids Research</em>, gkw982.
</div>
<div id="ref-osuna2020seminavis" class="csl-entry">
Osuna-Cruz, Cristina Maria, Gust Bilcke, Emmelien Vancaester, Sam De
Decker, Atle M Bones, Per Winge, Nicole Poulsen, et al. 2020. <span>“The
Seminavis Robusta Genome Provides Insights into the Evolutionary
Adaptations of Benthic Diatoms.”</span> <em>Nature Communications</em>
11 (1): 1–13.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
